tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -0,0 +1,379 @@
+        internal static void AgentStatus2Error<TInner>(TInner activity, Span span)
+            where TInner : IActivity
+        {
+            if (activity is IActivity6 { Status: ActivityStatusCode.Error } activity6)
+            {
+                span.Error = true;
+                // First iterate through Activity events first and set error.msg, error.type, and error.stack
+                if (span.GetTag(Tags.ErrorMsg) is null)
+                {
+                    span.SetTag(Tags.ErrorMsg, activity6.StatusDescription);
+                }
+            }
+            else if (span.GetTag("otel.status_code") == "STATUS_CODE_ERROR")
+            {
+                span.Error = true;
+                // First iterate through Activity events first and set error.msg, error.type, and error.stack
+                if (span.GetTag(Tags.ErrorMsg) is null)
+                {
+                    if (span.GetTag("otel.status_description") is string statusDescription)
+                    {
+                        span.SetTag(Tags.ErrorMsg, statusDescription);
+                    }
+                    else if (span.GetTag("http.status_code") is string statusCodeString)
+                    {
+                        if (span.GetTag("http.status_text") is string httpTextString)
+                        {
+                            span.SetTag(Tags.ErrorMsg, $"{statusCodeString} {httpTextString}");
+                        }
+                        else
+                        {
+                            span.SetTag(Tags.ErrorMsg, statusCodeString);
+                        }
+                    }
+                }
+            }
+        }
+
+        // See trace agent func spanKind2Type: https://github.com/DataDog/datadog-agent/blob/67c353cff1a6a275d7ce40059aad30fc6a3a0bc1/pkg/trace/api/otlp.go#L621
+        internal static string AgentSpanKind2Type(ActivityKind kind, Span span) => kind switch
+        {
+            ActivityKind.Server => SpanTypes.Web,
+            ActivityKind.Client => span.GetTag("db.system") switch
+            {
+                "redis" or "memcached" => "cache",
+                string => "db",
+                _ => "http",
+            },
+            _ => SpanTypes.Custom,
+        };
+
+        internal static string ComposeArrayString(Array array)
+        {
+            if (array.Length == 0)
+            {
+                return "[]";
+            }
+
+            StringBuilder sb = new();
+            sb.Append('[');
+            for (int i = 0; i < array.Length; i++)
+            {
+                var value = array.GetValue(i);
+                if (value is null)
+                {
+                    sb.Append("null");
+                }
+                else if (value is string s)
+                {
+                    sb.Append('"');
+                    sb.Append(s);
+                    sb.Append('"');
```

Comment:
Do we have to worry about escaping here? e.g. if the string is `some "quote" string`. I assume not, just wondering what otlp does etc?

Reply:
We might need to do something about this 🤔 I'll check what the .NET OTLP exporter does and make sure it matches

Reply:
If this is meant to be a json array, we probably want to use a json serializer to let it handle all the escaping, etc.

Reply:
I removed this method and replaced it with a JSON serializer in 0f685bce5bf6fdab3c04c2a3ca5a84c4dfd8b08a